web-admin/src/features/projects/github/GithubRepoSelectionDialog.svelte

```diff
@@ -0,0 +1,140 @@
+<script lang="ts">
+  import {
+    createAdminServiceGetProject,
+    createAdminServiceUpdateProject,
+    getAdminServiceGetGithubUserStatusQueryKey,
+    getAdminServiceGetProjectQueryKey,
+    type RpcStatus,
+  } from "@rilldata/web-admin/client";
+  import { GithubReposConnection } from "@rilldata/web-admin/features/projects/github/GithubReposConnection";
+  import {
+    AlertDialog,
+    AlertDialogContent,
+    AlertDialogDescription,
+    AlertDialogFooter,
+    AlertDialogHeader,
+    AlertDialogTitle,
+    AlertDialogTrigger,
+  } from "@rilldata/web-common/components/alert-dialog";
+  import { Button } from "@rilldata/web-common/components/button";
+  import Github from "@rilldata/web-common/components/icons/Github.svelte";
+  import Select from "@rilldata/web-common/components/forms/Select.svelte";
+  import Spinner from "@rilldata/web-common/features/entity-management/Spinner.svelte";
+  import { EntityStatus } from "@rilldata/web-common/features/entity-management/types";
+  import { eventBus } from "@rilldata/web-common/lib/event-bus/event-bus";
+  import { queryClient } from "@rilldata/web-common/lib/svelte-query/globalQueryClient";
+  import { invalidateRuntimeQueries } from "@rilldata/web-common/runtime-client/invalidation";
+  import type { AxiosError } from "axios";
+
+  export let open = false;
+  export let currentUrl: string;
+  export let project: string;
+  export let organization: string;
+
+  let githubUrl = currentUrl;
+  const githubReposConnection = new GithubReposConnection();
+  const githubRepos = githubReposConnection.userRepos;
+  const status = githubReposConnection.status;
+  const projectQuery = createAdminServiceGetProject(organization, project);
+
+  $: repoSelections =
+    $githubRepos.data?.repos?.map((r) => ({
+      value: r.url,
+      label: `${r.owner}/${r.name}`,
+    })) ?? [];
+
+  const updateProject = createAdminServiceUpdateProject();
+  async function updateGithubUrl() {
+    await $updateProject.mutateAsync({
+      name: project,
+      organizationName: organization,
+      data: {
+        githubUrl,
+      },
+    });
+    eventBus.emit("notification", {
+      message: `Set github repo to ${githubUrl}`,
+      type: "success",
+    });
+    void queryClient.refetchQueries([
+      getAdminServiceGetGithubUserStatusQueryKey(),
+      getAdminServiceGetProjectQueryKey(organization, project),
+    ]);
+    void invalidateRuntimeQueries(
+      queryClient,
+      $projectQuery.data.prodDeployment.runtimeInstanceId,
+    );
+    open = false;
+  }
+
+  function handleVisibilityChange() {
+    if (document.visibilityState !== "visible") return;
+    void githubReposConnection.focused();
```

Comment:
I had to read this function implementation to know what it was doing. If we put the Query Observer in this component, we could use `$githubUserReposQuery.refetch()`, which would be more explicit.